firefox-webext/background.js

```diff
@@ -2,9 +2,98 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+var fetchURL = 'https://api.github.com/repos/webcompat/web-bugs/issues?state=open&page=1&per_page=100';
+var issuesList = [];
+var requestIssues = [];
+var ignoreList = ['about'];
+var ts = Math.round(new Date().getTime() / 1000);
+var tsRefresh = 0;
 var prefix = 'https://webcompat.com/?open=1&url=';
 var screenshotData = '';
 
+function addRefresh(now) {
+  return (now + (24 * 3600));
+}
+
+function checkRefresh (){
+  var now = Math.round(new Date().getTime() / 1000);
+  if (issuesList.length == 0 || tsRefresh <= now) {
+    console.log('Data needs refresh.');
+    tsRefresh = addRefresh(now);
+    getIssues();
+    return true;
+  } else {
+    console.log('Data up to date.');
+    return false;
+  }
+}
+
+function getIssues() {
+  var now = Math.round(new Date().getTime() / 1000);
+  tsRefresh = addRefresh(now);
+  fetchBatch(fetchURL);
+}
+
+function fetchBatch(url){
+  fetch(url).then(function(response) {
```

Comment:
I wonder if this first `then` is doing anything... the next `then` takes the `response` (as `obj`) then does stuff with it. Maybe you can remove the first `then`, and pass in `response` and `response.json()` it?

(if that makes sense...)